decide_designable.py

```diff
@@ -74,14 +74,22 @@ def nussinov(sequence,level:int):
     return False
 
 def generate_sequence(g: Graph, v: int) -> list[str]:
+    pairs = [("A", "U"), ("U", "A"), ("C", "G"), ("G", "C")]
+    letters = ['A', 'U', 'C', 'G']
 
+    outputs = []
     if g.neighborhood_size(v, 1, "out") == 1:
-        return ["AU", "UA", "CG", "GC"]
+        # children dots
+        for letter in letters:
+            output = letter * g.vs[v]["unpaired_count_0"]
+            for pair in pairs:
+                outputs.append(pair[0] + output + pair[1])
+
+        outputs = add_dots(g, v, outputs)
+        return outputs
 
-    outputs = []
     prev_outputs = [""]
 
-    pairs = [("A", "U"), ("U", "A"), ("C", "G"), ("G", "C")]
     new_prev_outputs = []
     for u in g.neighbors(v, "out"):
```

Comment:
poniżej skorzystać z produktu kartezjanskiego.
`prev_outputs = [""] for u in g.neighbors(v, "out"): # budujemy sekwencje od lewej strony - zaczynając od skrajnego lewego dziecka itd, # product daje nam iloczyn kartezjański dotychczas zbudowanych sekwencji # i sekwencji nowych (zbudowanych dla kolejnego wierzchołka u) prev_outputs = [prev_output + new_output for prev_output, new_output in product(prev_outputs, generate_sequence(g, u))]`